### ISSUE TITLE
Could you please make it more obvious when imagefolder is loading hidden folders?

### ISSUE THREAD

**Issue by cynthia-rempel** [pytorch/vision#2831]
Thanks for making this dataloader! I was using imagefolder as part of a dataloader, then I was trying to use the dcgan example on the same folder with the images, and I started getting mysterious errors about the last index number on my cnn... It turned out that something made .ipynb_checkpoints folders in the imagefolder (I suspect the dcgan example). While this is clearly an example of user error, it would have saved a couple of hours of debugging if the imagefolder dataloader gave a warning about using hidden folders as a dataclass... I really do enjoy this amazing dataloader...

**Comment by pmeier** [comment]
@vfdev-5 I don't think this is a bug since the documentation doesn't state that hidden directories are excluded. 

@cynthia-rempel Nevertheless, IMO this is a reasonable assumption for the future. We could introduce a `is_valid_dir` parameter (similar to [`is_valid_file`](https://github.com/pytorch/vision/blob/a9c78f1b017d912fa9583ae6d93bff0a3807c29b/torchvision/datasets/folder.py#L209-L210)) that currently accepts any directory. It could emit a warning, that in the future hidden directories will be ignored by default. @fmassa?

**Comment by vfdev-5** [comment]
@pmeier if it is intended behaviour with hidden folders, than docs should clearly state about that, same as for symlinks.

**Comment by pmeier** [comment]
@vfdev-5 I don't know (but I can't imagine) that it is intended behavior. Still, some people might use it that way and thus we can't simply change it. That is why I propose to add `is_valid_dir` and phase out default support for hidden directories.

**Comment by m-Just** [comment]
I also encountered this issue but in my case the .ipynb_checkpoints folders were inside some of the class folders, and images inside these .ipynb_checkpoints folders were treated as new images of the affected classes. This behavior is much harder to notice than using hidden folders as dataclasses (mentioned by @cynthia-rempel) and can affect model training/testing silently for a long period of time.

**Comment by NicolasHug** [comment]
Since this seems to be a common mistake, it might be worth adding a `.. note::` in our docs indicating that all folders will be browsed, hidden or not.
